docs/reference/contributing/connectivity/WifiInterface.md

```diff
@@ -1,2 +1,43 @@
-## WiFiInterface
+<h2 id="wifi-port">Wi-Fi</h2>
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/wifi.png)<span>Wi-Fi driver</span></span>
+
+This document describes how to port and test an IEEE 802.11 Wi-Fi driver to Mbed OS. There are two types of Wi-Fi drivers in Mbed OS, depending on which protocol layer it implements. Wi-Fi drivers are either a special case of Ethernet driver or external IP stacks. Wi-Fi drivers require configuration from an application and, therefore, implement both the low-level EMAC API or network stack API and the high-level controlling interface API called `WiFiInterface`.
+
+### WifiInterface
+
+The `WifiInterface` class is the controlling API from application to driver. You can use this API for configuring security settings, network names and keys. You can also use this API for instantiating connection or disconnection phases from the network.
+
+Whether the driver is an AT-command driven external IP stack or an Ethernet device, this is the API to implement.
+
+The driver takes care of the network joining process and presents Mbed OS only Ethernet layer interface, or IP stack. The driver and the device itself do all of the security handshakes.
+
+### EMAC driver
+
+If the device is implementing Ethernet MAC, the driver provides the EMAC interface for Mbed OS. In this case, please refer to the [Ethernet MAC (EMAC) drivers](ethernet-port.html) porting guide first.
+
+### External IP stack
+
+If the device is an eternal IP stack, for example an AT-command driven module, it needs to implement the full `NetworkStack` API. Please refer to the [networkStack porting guide](networkstack.html) for implementation details.
+
+### Assumptions
```

Comment:
Thanks for adding these section. I like how this porting reference fits in with the others.